### PR TITLE
rsmangler: init at 1.5-unstable-2019-07-24

### PR DESCRIPTION
**Pull request by D3vil0p3r**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/rs/rsmangler/package.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  version = "1.5-unstable-2019-07-24";
+
+  src = fetchFromGitHub {
+    owner = "digininja";
+    repo = "RSMangler";
+    rev = "e85da7d4a6e6241a92389aecf376077adc7544c3";
+    hash = "sha256-DN20XzrlkunLyk4nkgytUJEtCOlFjWUUUAQ416l3Aug=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
```
Comment (a-n-n-a-l-e-e) [inline pkgs/by-name/rs/rsmangler/package.nix:19]:
```suggestion
  buildInputs = [ ruby ];
```

File: pkgs/by-name/rs/rsmangler/package.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{ lib
+, stdenvNoCC
+, fetchFromGitHub
+, makeWrapper
```
Comment (a-n-n-a-l-e-e) [inline pkgs/by-name/rs/rsmangler/package.nix:4]:
```suggestion
```

File: pkgs/by-name/rs/rsmangler/package.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    repo = "RSMangler";
+    rev = "e85da7d4a6e6241a92389aecf376077adc7544c3";
+    hash = "sha256-DN20XzrlkunLyk4nkgytUJEtCOlFjWUUUAQ416l3Aug=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  postPatch = ''
+    substituteInPlace rsmangler.rb \
+      --replace ./rsmangler.rb rsmangler
```
Comment (a-n-n-a-l-e-e) [inline pkgs/by-name/rs/rsmangler/package.nix:23]:
```suggestion
      --replace-quiet ./rsmangler.rb rsmangler \
      --replace-quiet rsmangler.rb rsmangler
```

File: pkgs/by-name/rs/rsmangler/package.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{bin,share/rsmangler}
+    cp rsmangler.rb $out/share/rsmangler/
+    makeWrapper ${ruby}/bin/ruby $out/bin/rsmangler \
+      --add-flags $out/share/rsmangler/rsmangler.rb
+    runHook postInstall
+  '';
```
Comment (a-n-n-a-l-e-e) [inline pkgs/by-name/rs/rsmangler/package.nix:33]:
```suggestion
  postInstall = ''
    install -Dm555 rsmangler.rb $out/bin/rsmangler
  '';
```

File: pkgs/by-name/rs/rsmangler/package.nix [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+  meta = with lib; {
+    description = "Perform various manipulations on the wordlists";
+    homepage = "https://github.com/digininja/RSMangler";
+    license = licenses.cc-by-sa-20;
+    mainProgram = "rsmangler";
+    maintainers = with maintainers; [ d3vil0p3r ];
+    platforms = platforms.unix;
```
Comment (a-n-n-a-l-e-e) [inline pkgs/by-name/rs/rsmangler/package.nix:41]:
```suggestion
    platforms = ruby.meta.platforms;
```

**Comment by a-n-n-a-l-e-e** [comment]
having a test would be nice too. from `testers` can add `testVersion`. like this.
https://nixos.org/manual/nixpkgs/unstable/#tester-testVersion
```nix
  passthru.tests.version = testers.testVersion {
    package = finalAttrs.finalPackage;
    command = "rsmangler --help";
    version = "rsmangler v ${lib.versions.majorMinor finalAttrs.version}";
  };
```

**Comment by D3vil0p3r** [comment]
> having a test would be nice too. from `testers` can add `testVersion`. like this. https://nixos.org/manual/nixpkgs/unstable/#tester-testVersion
> 
> ```nix
>   passthru.tests.version = testers.testVersion {
>     package = finalAttrs.finalPackage;
>     command = "rsmangler --help";
>     version = "rsmangler v ${lib.versions.majorMinor finalAttrs.version}";
>   };
> ```

This code related to tests must be implemented in `package.nix`?

**Comment by a-n-n-a-l-e-e** [comment]
> This code related to tests must be implemented in `package.nix`?

yes -- just add it to the derivation just above the `meta` section. you'll need to use the finalAttrs pattern as well as add `testers` to the arguments.  there are lots of examples in nixpkgs if you recursive grep for `passthru.tests.version`

**Comment by D3vil0p3r** [comment]
@a-n-n-a-l-e-e fixed.

**Review by ghost — APPROVED** [review]

**Pull request merged**